### PR TITLE
Advanced search page

### PR DESCRIPTION
**Pull request by sonalkr132**
screenshot from 2017-04-17 21-35-10
---
screenshot from 2017-04-17 21-30-49
---
screenshot from 2017-04-17 21-32-47

### CODE REVIEW

File: app/assets/javascripts/search.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+var $main        = $('#home_query'),
+    $name        = $('input#name'),
+    $summary     = $('input#summary');
```
Comment (simi) [inline app/assets/javascripts/search.js]: Use `var $summary = $('input#summary');` or `,` on the end of this (and following lines).

File: app/assets/javascripts/search.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    $description = $('input#description');
+    $downloads   = $('input#downloads');
+    $updated     = $('input#updated');
+
+$name.add($summary)
+  .add($description)
+  .add($downloads)
+  .add($updated)
+  .bind('keyup change', function(e) {
```
Comment (simi) [inline app/assets/javascripts/search.js]: I think we can use `oninput` event here.

File: app/assets/javascripts/search.js (hunk 1 above)
Comment (simi) [inline app/assets/javascripts/search.js]: Use `var $summary = $('input#summary');` or `,` on the end of this (and following lines). 

PS: *I prefer explicit var.*

File: app/views/searches/advanced.html.erb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<% @title = t("advanced_search") %>
```
Comment (simi) [inline app/views/searches/advanced.html.erb]: Maybe set in controller?

Comment (sonalkr132) [inline app/views/searches/advanced.html.erb]: Setting `@title` in view seems to be a [convention followed in all of the templates](https://github.com/rubygems/rubygems.org/blob/master/app/views/rubygems/show.html.erb#L1).

**Comment by sonalkr132** [comment]
@jvanbaarsen @simi Thanks for review :v: 
Updated!

File: app/assets/javascripts/search.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  .add($updated)
+  .on('input', function(e) {
+    var name        = $name.val().length > 0 ? 'name: ' + $name.val() : '';
+    var summary     = $summary.val().length > 0 ? 'summary: ' + $summary.val() : '';
+    var description = $description.val().length > 0 ? 'description: ' + $description.val() : '';
+    var downloads   = $downloads.val().length > 0 ? 'downloads: ' + $downloads.val() : '';
+    var updated     = $updated.val().length > 0 ? 'summary: ' + $updated.val() : '';
+
+    $main.val($.trim(name + ' ' + summary + ' ' + description + ' ' + downloads + ' ' + updated));
+});
```
Comment (dwradcliffe) [inline app/assets/javascripts/search.js]: All this javascript should only be run on the advanced search page. So check for the search class or something.

File: app/controllers/searches_controller.rb [hunk 5]
```diff
@@ -8,6 +8,8 @@ def show
     redirect_to rubygem_path(@exact_match) if @exact_match && @gems.size == 1
   end
 
+  def advanced; end
```
Comment (dwradcliffe) [inline app/controllers/searches_controller.rb]: nitpick: I'd rather this be 2 lines.

**Comment by sonalkr132** [comment]
Updated!

**Comment by sonalkr132** [comment]
Ping!
6a00d8341d3df553ef0148c80ac6ef970c-800wi

**Comment by segiddins** [comment]
@sonalkr132 is this deployed somewhere I could try it out?

**Comment by sonalkr132** [comment]
Not yet. I am waiting for one more approval before it goes into staging.
You can try it locally, you don't need elasticsearch to test changes in this PR.

File: app/views/searches/advanced.html.erb [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    <dd><%= text_field_tag :summary, "", placeholder: "ORM, NoSQL", class: "form__input"%></dd>
+
+    <dt><%= label_tag :description, t(".description"), class: "form__label" %></dt>
+    <dd><%= text_field_tag :description, "", placeholder: "associations AND validations", class: "form__input"%></dd>
+
+    <dt><%= label_tag :downloads, t(".downloads"), class: "form__label" %></dt>
+    <dd><%= text_field_tag :downloads, "", placeholder: "> 20,000", class: "form__input"%></dd>
+
+    <dt><%= label_tag :updated, t(".updated"), class: "form__label" %></dt>
+    <dd><%= text_field_tag :updated, "", placeholder: "> YYYY-MM-DD, YYYY-MM-DD", class: "form__input"%></dd>
```
Comment (simi) [inline app/views/searches/advanced.html.erb]: What about to show current or some fixed date in placeholder instead of this pattern?

Comment (sonalkr132) [inline app/views/searches/advanced.html.erb]: Updated to use `1.week.ago.strftime('%F')`

**Pull request merged**